azurelinuxagent/common/protocol/wire.py

```diff
@@ -780,45 +779,40 @@ def update_host_plugin_from_goal_state(self):
+    def update_goal_state(self, force_update=False):
         """
         Updates the goal state if the incarnation or etag changed or if 'force_update' is True
         """
         try:
             #
-            # The entire goal state needs to be retrieved from the WireServer (via the GoalState class), and the HostGAPlugin
+            # The goal state needs to be retrieved using both the WireServer (via the GoalState class) and the HostGAPlugin
             # (via the self._fetch_vm_settings_goal_state method).
             #
-            # We fetch it in 3 parts:
+            # We always need at least 2 queries: one to the WireServer (to check for incarnation changes) and one to the HostGAPlugin
+            # (to check for extension updates).
             #
-            # 1) The "main" goal state from the WireServer, which includes the incarnation, container ID, role config, and URLs
-            #    to the rest of the goal state (certificates, remote users, extensions config, etc). We do this first because
-            #    we need to initialize the HostGAPlugin with the container ID and role config.
+            # We start by fetching the goal state from the WireServer. The response to this initial query will include the incarnation,
+            # container ID, role config, and URLs to the rest of the goal state (certificates, remote users, extensions config, etc). We
+            # do this first because we need to initialize the HostGAPlugin with the container ID and role config.
             #
             goal_state = GoalState(self)
 
             self._update_host_plugin(goal_state.container_id, goal_state.role_config_name)
 
             #
-            # 2) Then we fetch the vmSettings from the HostGAPlugin. We do this before fetching the rest of the goal state from the
-            #    WireServer to minimize the time between the initial call to the WireServer and the call to the HostGAPlugin (and hence
-            #    reduce the window in which a new goal state may arrive in-between the 2 calls)
+            # Then we fetch the vmSettings from the HostGAPlugin; the response will include the goal state for extensions.
             #
             vm_settings_goal_state, vm_settings_goal_state_updated = (None, False)
 
             if conf.get_enable_fast_track():
                 try:
                     vm_settings_goal_state, vm_settings_goal_state_updated = self._fetch_vm_settings_goal_state(force_update=force_update)
-
-                except Exception as error:
-                    # _fetch_vm_settings_goal_state() does its own detailed error reporting and raises ProtocolError; do not report those
-                    if not isinstance(error, ProtocolError):
-                        self._vm_settings_error_reporter.report_error(format_exception(error))
-                self._vm_settings_error_reporter.report_summary()
+                except VmSettingsNotSupported:
```

Comment:
In case of a `ProtocolError`, we won't even try to fetch the `ExtensionsConfig` from Wireserver. Is that the expected behavior? Shouldn't we try to fetch ExtensionsConfig from Wireserver if we can't fetch VMSettings for whatever reason?

Reply:
This change makes VmSettings *the* way to get the extensions goal state. Errors on vmSettings need to be handled the same as  any other errors in the WireServer goal state.

Reply:
Gotcha, basically in the scenario that there is FT data in the vmSettings, then falling back to Wireserver would be wrong too. 

Thanks for explaining!

Reply:
`Gotcha, basically in the scenario that there is FT data in the vmSettings, then falling back to Wireserver would be wrong too.
`

Yes, exactly. Not for this specific PR because everything is Fabric still, but well get FastTrack as soon as I implement the handshake with CRP (coming 2 or 3 PRs from this one)